src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -205,14 +211,13 @@ void BaseCouplingScheme::receiveData(const m2n::PtrM2N &m2n, const DataMap &rece
 void BaseCouplingScheme::receiveDataForWindowEnd(const m2n::PtrM2N &m2n, const DataMap &receiveData)
 {
   // buffer current time
-  const double oldTime = getTime();
+  const impl::TimeHandler oldTime = _time;
   // set _time state to point to end of this window such that _time in receiveData is at end of window
-  _time(_nextTimeWindowSize);
+  _time.progressBy(_nextTimeWindowSize);
   // receive data for end of window
   this->receiveData(m2n, receiveData);
   // reset time state;
-  _time = KahanAccumulator{};
-  _time(oldTime);
+  _time = oldTime;
 }
```

Comment:
This part with `oldTime` still looks strange, but is necessary due to the way how `receiveData` works, if we set `substeps="False"`. Maybe put a comment here that `oldTime` is something temporary. An issue on this topic would be helpful such that I don't have to write this again in every PR review (I know: I should do this and can take care of it).

Reply:
I tried to summarize the big picture here: https://github.com/precice/precice/issues/1957
@fsimonis would you mind adding a comment with a pointer to https://github.com/precice/precice/issues/1957 in this PR? I hope the issue is somehow understandable. Unfortunately, the situation is confusing. But fixing it needs some effort with low benefit (from the short term perspective).

Reply:
Done